### PR TITLE
/upgrade only updates existent files

### PR DESCRIPTION
**Pull request by dianabarsan**
- `/upgrade` now only overwrites existing files. 
- if a non-existent file is passed to `/upgrade`, writing and validating will be skipped, and response is changed to `{ ok: false }`
- adds `/install` endpoint that can also add new docker-compose files to the folder

#9

### CODE REVIEW

File: src/containers.js [hunk 1]
```diff
@@ -19,20 +19,25 @@ const overwriteComposeFile = async (filePath, fileContents) => {
...
 };
 
-const upgrade = async (fileName, fileContents) => {
+const update = async (fileName, fileContents, install = false) => {
   if (!fileName) {
     throw new Error('Invalid docker-compose file name');
   }
 
+  const filePath = path.join(dockerComposeFilePath, fileName);
+  if (install === fs.existsSync(filePath)) {
```
Comment (garethbowen) [inline src/containers.js:28]: Use `await fs.exists` instead. It's already an async method so it's not adding any complexity, and it's better for Node not to block on the filesystem.

File: src/server.js [hunk 2]
```diff
@@ -18,8 +18,8 @@ const upgrade = async (req, res) => {
 
   try {
     for (const [fileName, fileContents] of payload) {
-      await containers.upgrade(fileName, fileContents);
-      response[fileName] = { ok: true };
+      const updated = await containers.update(fileName, fileContents, install);
+      response[fileName] = { ok: updated };
```
Comment (garethbowen) [inline src/server.js:22]: Maybe include a reason if `ok: false`, ie:

```json
{ "ok": false, "reason": "Existing installation not found. Use '/install' API to install." }
```

**Review by garethbowen — CHANGES_REQUESTED** [review]
Very nice, thanks!

Two minor suggestions.

File: src/containers.js (hunk 1 above)
Comment (dianabarsan) [reply src/containers.js:28]: There is no async `fs.exists`. There's a long explanation about why [in the callback `exists` documentation](https://nodejs.org/dist/latest-v16.x/docs/api/fs.html#fsexistspath-callback). I can use `stat`, but it complicates the code somewhat.

Comment (garethbowen) [reply src/containers.js]: Oh interesting.  TIL `fs.exists` is deprecated, but `fs.existsSync` is not. That's so backwards. Let's roll with it for now.

Comment (dianabarsan) [reply src/containers.js]: i already changed it to stat :)

**Review by garethbowen — APPROVED** [review]
Awesome!

Thanks for taking the time to use `stat`!

**Pull request merged**